### PR TITLE
tools: txn replayer

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

The debugging tool allows to catchup to a certain round and either stop or start applying transaction(s) from a file, directory, txlog files or directory.

It uses `ledger.OpenLedger` + `pools.TransactionPool` while adding new data into ledger, and `node.follower_node` for catchup.

## Test Plan

Tested manually

### CODE REVIEW

File: tools/debug/transplanter/main.go [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+		os.Exit(1)
+	}
+
+	if len(*txnDir) > 0 && len(*txnLogDir) > 0 {
+		fmt.Fprintln(os.Stderr, "Cannot specify both transactions and transaction logs dirs")
+		os.Exit(1)
+	}
+
+	if !*force {
+		fmt.Println("Running this command could damage your node installation, proceed anyway (N/y)?")
```
Comment (iansuvak) [inline tools/debug/transplanter/main.go:351]: What is actually at risk of being broken by running this tool?

File: tools/debug/transplanter/main.go [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+	Ts   time.Time
+	IP   string
+	Port int
+}
+
+type headerDecoder interface {
+	decodeHeader(*snappy.Reader) (*headerRow, int, error)
+}
+
+type decoderV1 struct{}
```
Comment (iansuvak) [inline tools/debug/transplanter/main.go:163]: we could probably remove decoderv1 but no harm in keeping it

**Review by iansuvak — APPROVED** [review]
Generally looks good and safe to merge especially since it's a standalone tool.

**Comment by bbroder-algo** [comment]
How do you generate the transaction files?

**Comment by algorandskiy** [comment]
> How do you generate the transaction files?

This is up to the user. One might run a node and dump all validated txns, other might extract them from a block.

File: tools/debug/transplanter/main.go (hunk 1 above)
Comment (algorandskiy) [reply tools/debug/transplanter/main.go:351]: yon cannot reuse this node dir after such a custom block committed. I think I need to change the message ask ensure there is a backup copy. For example, catching up mainnet might take hours, and can be easily spoiled by running this tool. Not a big deal for one and a problem for others (say archival node dirs).

**Review by bbroder-algo — APPROVED** [review]

**Pull request merged**